projects/storefrontlib/src/layout/main/storefront.component.ts

```diff
@@ -14,9 +27,30 @@ export class StorefrontComponent implements OnInit, OnDestroy {
   @HostBinding('class.start-navigating') startNavigating;
   @HostBinding('class.stop-navigating') stopNavigating;
 
+  // required by esc focus
+  @HostBinding('tabindex') tabindex = '0';
+
+  @ViewChild(SkipLinkComponent) child: SkipLinkComponent;
+
+  private config: EscapeFocusConfig = {
+    focusOnEscape: true,
+    focusOnDoubleEscape: true,
+  };
+
+  @HostListener('keydown.escape', ['$event'])
+  protected handleEscape(event: KeyboardEvent): void {
```

Comment:
the reason is that I didn't want to pollute the storefront host element with another nested element, so that I could apply the directive. Unfortunately there's  no way to apply a directive in the controller logic, only goes through elements. This was one of the reasons to keep directives clean so we can use it like this.